plugins/cros-ec/fu-cros-ec-usb-device.c

```diff
@@ -578,6 +701,24 @@ fu_cros_ec_usb_device_prepare_firmware (FuDevice *device,
 	return g_steal_pointer (&firmware);
 }
 
+static gboolean
+fu_cros_ec_usb_device_detach (FuDevice *self, GError **error)
+{
+	g_debug ("cros_ec_usb_device_detach");
+
+	if (!fu_cros_ec_usb_device_reset_to_ro (self, error)) {
+
+		return FALSE;
+	}
+/*
```

Comment:
if this is going to replug the device, you want to do something like:

    fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
    fu_device_add_flag (device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);

If it's coming back with the same VID/PID then that's fine, otherwise you'll need to add a "counterpart GUID" of the new VID/PID so that the daemon knows what it's waiting for before ->write_firmware gets called.

Reply:
You'll probably want to also use
```
	fu_device_add_flag (device, FWUPD_DEVICE_FLAG_IS_BOOTLOADER);
```

When in RO mode (and if it's the same VID/PID remove that flag when going back to RW)

Reply:
Alright! I will give that a try. Thanks so much for the pointer to those state changes.

Reply:
And yes, the VID/PID will be the same.

Reply:
Thanks @hughsie and @superm1.

Setting those flags makes all the difference. I'm able to get it to go around the steps again with the detatch. I'll make sure to implement attach as well to reboot into RW after an update.

If that part works, I'll try ANOTHER_WRITE_REQUIRED.